api/src/main/java/com/okta/idx/sdk/api/wrapper/AuthenticationWrapper.java

```diff
@@ -164,72 +184,617 @@ public static AuthenticationResponse authenticate(IDXClient client, Authenticati
+                try {
+                    enrollRemediationOption = extractRemediationOption(remediationOptions, RemediationType.SKIP);
+                } catch (IllegalArgumentException e) {
+                    logger.warn("Skip authenticator not found in remediation option");
+                    enrollRemediationOption = extractRemediationOption(remediationOptions, RemediationType.SELECT_AUTHENTICATOR_ENROLL);
+                }
+            }
+        } catch (ProcessingException e) {
+            logger.error("Error", e);
+        } catch (IllegalArgumentException e) {
+            logger.error("Exception occurred", e);
+        }
+
+        authenticationResponse.setIdxClientContext(idxClientContext);
+        return authenticationResponse;
+    }
+
+    /**
+     * Skip optional authenticator enrollment.
+     *
+     * @param client                the IDX Client
+     * @param idxClientContext      the IDC Client context
+     * @return the Authentication response
+     */
+    public static AuthenticationResponse skipAuthenticatorEnrollment(IDXClient client, IDXClientContext idxClientContext) {
+
+        AuthenticationResponse authenticationResponse = new AuthenticationResponse();
+
+        try {
+            IDXResponse introspectResponse = client.introspect(idxClientContext);
+            String stateHandle = introspectResponse.getStateHandle();
+
+            RemediationOption[] remediationOptions = introspectResponse.remediation().remediationOptions();
+
+            RemediationOption remediationOption = extractRemediationOption(remediationOptions, RemediationType.SKIP);
+
+            SkipAuthenticatorEnrollmentRequest skipAuthenticatorEnrollmentRequest = SkipAuthenticatorEnrollmentRequestBuilder.builder()
+                    .withStateHandle(stateHandle)
+                    .build();
+
+            remediationOption.proceed(client, skipAuthenticatorEnrollmentRequest);
+
+        } catch (ProcessingException e) {
+            logger.error("Error occurred", e);
+            Arrays.stream(e.getErrorResponse().getMessages().getValue()).forEach(msg -> authenticationResponse.addError(msg.getMessage()));
+            logger.error("Something went wrong! {}, {}", e, authenticationResponse.getErrors());
+        } catch (IllegalArgumentException e) {
+            logger.error("Exception occurred", e);
+            authenticationResponse.addError(e.getMessage());
+        }
+
+        authenticationResponse.setIdxClientContext(idxClientContext);
+        return authenticationResponse;
+    }
+
+    /**
+     * Helper to populate the UI options to be shown on Authenticator options page.
+     *
+     * @param client                the IDX Client
+     * @param idxClientContext      the IDC Client context
+     * @return the list of {@link AuthenticatorUIOption} options
+     */
+    public static List<AuthenticatorUIOption> populateAuthenticatorUIOptions(IDXClient client, IDXClientContext idxClientContext) {
```

Comment:
Should these come from the response instead?

Reply:
Idea is same as https://github.com/okta/okta-idx-java/pull/47#discussion_r612568170